src/ray/core_worker/core_worker.cc

```diff
@@ -1650,39 +1646,45 @@ std::shared_ptr<rpc::RuntimeEnvInfo> CoreWorker::OverrideTaskOrActorRuntimeEnvIn
 
   if (options_.worker_type == WorkerType::DRIVER) {
     if (IsRuntimeEnvEmpty(runtime_env_info->serialized_runtime_env())) {
-      return job_runtime_env_info_;
+      return std::make_shared<rpc::RuntimeEnvInfo>(
+          worker_context_.GetCurrentJobConfig().runtime_env_info());
+    }
+
+    auto job_serialized_runtime_env =
```

Comment:
```suggestion
    const auto &job_serialized_runtime_env =
```

Reply:
GetCurrentJobConfig() returns by value due to thread safety issue.